### ISSUE TITLE
Support 'email_verified' for local connector

### ISSUE THREAD

**Issue by DusanKasan** [dexidp/dex#957]
When using the local connector, it could be beneficial to also store a boolean value for 'email_verified'. The fact Dex does not store this now is a problem when you are using the local connector and your registration flow contains the email verification. 

## Proposed changes
- Add `email_verified` boolean field to password table
- Update the gRPC API to reflect this
- Show an error on login when trying to log in as a non-verified user
  - Maybe this could be controlled via a configuration parameter? If so, should it work globally or per connector?

Does this change makes sense?

**Comment by rithujohn191** [comment]
Hi @DusanKasan ,

Thank you for your suggestion. Are you proposing that the actual email verification would be handled on your client's end and Dex would only store the Boolean info of if the email is actually verified?

**Comment by DusanKasan** [comment]
Yes, @rithujohn191. I think keeping just a boolean flag on dex and handling it from a separate client via the gRPC API is in the spirit of Dex 2.0 (just like registration is done). What do you think?

**Comment by rithujohn191** [comment]
That sounds very reasonable @DusanKasan, and I am sure it would benefit others too. I think we just need to make sure that this feature is not turned on by default. I would be more than happy to review a PR for this change.

**Comment by DusanKasan** [comment]
The relevant PR is https://github.com/coreos/dex/pull/960. Could you take a look?

Also a bit off-topic, what do you think about adding phone login via "phone_number" and "phone_number_verified" (standard OIDC claims) in a followup PR? Basically introduce a new "phone" connector that would be pretty much the same as "password" connector and would have the same gRPC API as for "passwords" and such.

**Comment by rithujohn191** [comment]
Thank you @DusanKasan. Sorry I have been swamped, I'll take a look at your PR soon.

Login via "Phone Number" is not really a popular request and adding another connector would also mean more work maintaining it, so unfortunately at this time we don't plan on adding that feature.

**Comment by ericchiang** [comment]
Said this on the PR, but I'd really not like to add more user management features to dex.

**Comment by nabokihms** [comment]
Was fixed in https://github.com/dexidp/dex/releases/tag/v2.45.0
